tests/puppeteer/campaigns/functional/BO/catalog/monitoring/01_deleteEmptyCategory.js

```diff
@@ -0,0 +1,137 @@
+require('module-alias/register');
+// Using chai
+const {expect} = require('chai');
+const helper = require('@utils/helpers');
+const files = require('@utils/files');
+const loginCommon = require('@commonTests/loginBO');
+// Importing pages
+const BOBasePage = require('@pages/BO/BObasePage');
+const LoginPage = require('@pages/BO/login');
+const DashboardPage = require('@pages/BO/dashboard');
+const CategoriesPage = require('@pages/BO/catalog/categories');
+const AddCategoryPage = require('@pages/BO/catalog/categories/add');
+const MonitoringPage = require('@pages/BO/catalog/monitoring');
+const CategoryFaker = require('@data/faker/category');
+
+let browser;
+let page;
+let numberOfCategories = 0;
+let numberOfEmptyCategories = 0;
+const createCategoryData = new CategoryFaker();
+
+// Init objects needed
+const init = async function () {
+  return {
+    boBasePage: new BOBasePage(page),
+    loginPage: new LoginPage(page),
+    dashboardPage: new DashboardPage(page),
+    categoriesPage: new CategoriesPage(page),
+    addCategoryPage: new AddCategoryPage(page),
+    monitoringPage: new MonitoringPage(page),
+  };
+};
+
+/*
+Create new category
+Check existence of new category in monitoring page
+Delete category and check deletion in category page
+ */
+describe('Create empty category and delete it from monitoring page', async () => {
+  // before and after functions
+  before(async function () {
+    browser = await helper.createBrowser();
+    page = await helper.newTab(browser);
+    this.pageObjects = await init();
+  });
+  after(async () => {
+    await helper.closeBrowser(browser);
+    /* Delete the generated image */
+    await files.deleteFile(`${createCategoryData.name}.jpg`);
+  });
+  // Login into BO and go to categories page
+  loginCommon.loginBO();
+
+  it('should go to catalog > categories page', async function () {
+    await this.pageObjects.boBasePage.goToSubMenu(
+      this.pageObjects.boBasePage.catalogParentLink,
+      this.pageObjects.boBasePage.categoriesLink,
+    );
+    await this.pageObjects.boBasePage.closeSfToolBar();
+    const pageTitle = await this.pageObjects.categoriesPage.getPageTitle();
+    await expect(pageTitle).to.contains(this.pageObjects.categoriesPage.pageTitle);
+  });
+
+  it('should reset all filters and get number of categories in BO', async function () {
+    numberOfCategories = await this.pageObjects.categoriesPage.resetAndGetNumberOfLines();
+    await expect(numberOfCategories).to.be.above(0);
```

Comment:
Is this test really needed? Aren't we able to detect the real number of rows?
or should be something like?
```suggestion
    await expect(numberOfCategories).to.be.at.least(4);
```
(the original number of categories after a fresh install or something similar)

Reply:
Yes we have at least 3 subcategories of home category, we can check that
The test is not really needed , what needed is to get the real number of categories to use it in other tests (after creation, after reset, after delete ...)

```js
   numberOfCategories = await this.pageObjects.categoriesPage.resetAndGetNumberOfLines();
```